### ISSUE TITLE
setIdChanged doesn't correctly handle element sources/targets

### ISSUE THREAD

**Issue by miloszw** [sporritt/jsPlumb#408]
It doesn't look like `setIdChanged` has an effect on the `sourceEndpointDefinitions` object which results in buggy behaviour when changing Id of elements which have been "`makeSource`d". Also, the `anchorManager` seems to be undefined.

https://jsfiddle.net/oap7u98n/1/

**Comment by miloszw** [comment]
Adding below lines [here](https://github.com/sporritt/jsPlumb/blob/master/src/jsPlumb.js#L2771) seems to helps

````javascript
this.sourceEndpointDefinitions[newId] = this.sourceEndpointDefinitions[id];
this.targetEndpointDefinitions[newId] = this.targetEndpointDefinitions[id];
delete this.sourceEndpointDefinitions[id];
delete this.targetEndpointDefinitions[id];
````

**Comment by sporritt** [comment]
@miloszw i think i have fixed this (and the makeTarget case) in the 1.7.7 branch
